ykcapi/src/lib.rs

```diff
@@ -124,36 +124,27 @@ struct AOTVar {
     bbidx: usize,
     instridx: usize,
     fname: *const i8,
+    sfidx: usize,
 }
 
-/// Address and length of a vector containing AOTVars. Mirrors the struct defined in
+/// Address and length of a vector. Mirrors the struct defined in
 /// ykllvmwrap/jitmodbuilder.cc.
 #[derive(Debug)]
 #[repr(C)]
-pub struct AOTMap {
+pub struct VecLen {
     addr: *const c_void,
     length: usize,
 }
 
-/// Location (basic block index, instruction index, function name) in the AOTModule where the guard
-/// failure occured. Mirrors the struct defined in ykllvmwrap/jitmodbuilder.cc.
-#[derive(Debug)]
-#[repr(C)]
-pub struct CurPos {
-    bbidx: usize,
-    instridx: usize,
-    fname: *const i8,
-}
-
 /// Reads the stackmap and saved registers from the given address (i.e. the return address of the
 /// deoptimisation call).
 #[cfg(target_arch = "x86_64")]
 #[no_mangle]
 pub extern "C" fn yk_stopgap(
     sm_addr: *const c_void,
     sm_size: usize,
-    aotmap: &AOTMap,
-    curpos: &CurPos,
+    aotmap: &VecLen,
+    actframes: &VecLen,
```

Comment:
What does `act` mean?

Reply:
Ah, "active". I get it.

Reply:
Active. Dunno why I named it this. Can probably just rename it to `frames`.